src/Driver/OCI8/Result.php

```diff
@@ -115,10 +120,23 @@ public function free(): void
      */
     private function fetch(int $mode)
     {
-        return oci_fetch_array(
-            $this->statement,
-            $mode | OCI_RETURN_NULLS | OCI_RETURN_LOBS
+        set_error_handler(
+            function (int $errno, string $errstr, string $errfile, int $errline): bool {
+                return Result::handleError($errno, $errstr, $errfile, $errline);
```

Comment:
You can move the implementation right here. It's not that big of a deal. You can also suppress the code style error (which is a false-positive) in `phpcs.xml`.